docs/concepts/activation.md

```diff
@@ -0,0 +1,321 @@
+in the [[hook] section of your manifest][hook-section].
+Since this script is run by the Bash shell we're using,
+you only need to worry about the synax and oddities of one shell when writing
+this script.
+This is convenient, but it comes with some tradeoffs.
+
+This Bash subshell is _non-interactive_.
+Some programs behave differently when they execute in a shell that's not
+interactive,
+and most shells will not (by default) expand aliases when run non-interactively.
+Also, remember that any functions or aliases you define in this Bash subshell
+via the `hook.on-activate` script will not be inherited by your shell later.
+
+Finally, we `exec` your shell with some overrides that allow us to inject our
+own configuration,
+such as `source`ing the scripts defined in the
+[[profile] section of your manifest][profile-section].
+Once those scripts have been sourced,
+we hand control back to you.
+Since the scripts in the `[profile]` section are sourced by your shell,
+this is where you can define aliases and functions that you'd like to be
+available in your activated environment.
+
+### hook vs. profile in a nutshell
+
+So, that was a lot of technical detail.
+To make life easier for you when it comes to writing scripts for your
+environment,
+here is some simple guidance:
+
+**hook.on-activate**:
+
+- Always Bash, so there's only one shell syntax and set of quirks to keep in mind.
+- Can't use aliases.
+- Can define functions to use within the hook, but they won't be passed down to other shells.
+- _Can_ define environment variables that need to be computed.
+
+**profile**:
+
+- Syntax depends on the shell.
+- Can define functions and aliases.
+- Can source scripts needed for other programs to work properly e.g. the `activate` script for a Python virtual environment.
+- _Can_ define environment variables that need to be computed.
+
+In short, it's probably best to put as much as you can in `hook.on-activate`
+until you have shell-specific needs, you need aliases, or you need to source
+a third-party script into your shell.
+
+## Attaching
+
+Everything we've discussed so far is about what happens when you start a new
+activation of an environment.
+However, if you're simply activating a second instance of an environment,
+all of the setup done in your `hook` and script will already have
+been done,
+so you would be doing the same exact thing.
+In addition, you would be setting all of the same exact environment variables
+from your `[vars]` `[hook]` sections as before.
+
+In short, this would be doing a lot of pointless work.
+For that reason, we record those environment variables and apply them to
+subsequent activations rather than computing them again.
+We call this "attaching" to an activation, and we do it automatically for you
+to make activation as fast as possible.
+
+```d2 scale="1.0"
+shape: sequence_diagram
+shell1: First shell
+shell2: Second shell
+files: \"Somewhere\"
```

Comment:
nit: I think "Somewhere" is confusing and if we just say save environment variables" that's probably enough information. I don't think we need a box that basically means "you don't care" because...they don't care.
```suggestion
files: \"Somewhere\"
```